Source/engine.cpp

```diff
@@ -145,7 +146,10 @@ void CelBlitSafeTo(CelOutputBuffer out, int sx, int sy, BYTE *pRLEBytes, int nDa
 			if (!(width & 0x80)) {
 				i -= width;
 				if (dst < out.end() && dst >= out.begin()) {
-					memcpy(dst, src, std::min(static_cast<ptrdiff_t>(width), out.end() - dst));
+					// GenerateItemLabelOffsets loops this function for all items, we don't actually want them on screen, just the min and max x position

```

Comment:
We probably want to avoid putting this into a general-purpose rendering function. Let's make a separate function for measuring instead?

Reply:
well, you measure by rendering :D

Reply:
That's fine but the rendering function shouldn't know anything about the particular game features

A new engine.cpp-internal function can be introduced and used in both the rendering and the measuring implementations

Reply:
This is really my only gripe with this implementation. I think we can probably carve something useful out of `CheckCursMove()`, so waiting for that to happen one way or another.

Reply:
You can find the width of the Cel sprite by simply going to the first 0x80; I'll add a function for you

Reply:
I've sent #1779, it's untested and I'm not sure how correct it is or whether it needs to handle single-frame CELs differently but it should be a good starting point.

Reply:
I see that this measures the count of non-transparent pixels.
This is *probably* not exactly what you want, you want the min/max x. Such a function should be easy to write as well.